packages/cyberstorm/src/components/MultiSelectSearch/MultiSelectSearch.tsx

```diff
@@ -0,0 +1,191 @@
+
+    function remove(incomingOption: MultiSelectSearchOption) {
+      onChange(value.filter((option) => option.value !== incomingOption.value));
+    }
+
+    const hideMenu = React.useCallback(
+      (e: MouseEvent | TouchEvent) => {
+        if (
+          menuRef.current &&
+          isVisible &&
+          !menuRef.current.contains(isNode(e.target) ? e.target : null)
+        ) {
+          setIsVisible(false);
+          onBlur();
+        }
+      },
+      [setIsVisible, isVisible, onBlur, menuRef]
+    );
+
+    // Event listeners for closing menu when clicking or touching outside.
+    React.useEffect(() => {
+      document.addEventListener("mousedown", hideMenu);
+      document.addEventListener("touchstart", hideMenu);
+      return () => {
+        document.removeEventListener("mousedown", hideMenu);
+        document.removeEventListener("touchstart", hideMenu);
+      };
+    });
+
+    React.useEffect(() => {
+      const updatedOptions = options.filter(
+        (option) =>
+          !value.includes(option) &&
+          option.label.toLowerCase().includes(search.toLowerCase())
+      );
+      setFilteredOptions(updatedOptions);
+    }, [value, search, options, setFilteredOptions]);
+
+    return (
+      <div className={styles.root} ref={ref}>
+        <div className={styles.selected}>
+          {value.map((option) => {
+            return (
+              <Button.Root
+                key={option.value}
+                onClick={() => remove(option)}
+                paddingSize="small"
+                style={{ gap: "0.5rem" }}
+              >
+                <Button.ButtonLabel>{option.label}</Button.ButtonLabel>
+                <Button.ButtonIcon>
+                  <FontAwesomeIcon icon={faXmark} />
+                </Button.ButtonIcon>
+              </Button.Root>
+            );
+          })}
+        </div>
+        <div
+          className={styles.search}
+          onFocus={(e) => {
+            inputRef.current && inputRef.current.focus();
+            e.stopPropagation();
+          }}
+          role="button"
+          tabIndex={0}
+          ref={menuRef}
+        >
+          <div className={styles.inputContainer} data-color={color}>
+            <input
+              className={styles.input}
+              value={search}
+              data-color={color}
+              onFocus={() => setIsVisible(true)}
+              onChange={(e) => setSearch(e.currentTarget.value)}
+              ref={inputRef}
+              placeholder={placeholder}
+            />
+            <button
+              onClick={(e) => {
+                setSearch("");
+                e.stopPropagation();
+              }}
+              className={styles.clearSearch}
+            >
+              <Icon inline>
+                <FontAwesomeIcon icon={faCircleXmark} />
+              </Icon>
+            </button>
+            <div className={styles.inputButtonDivider}></div>
+            <button
+              onClick={(e) => {
```

Comment:
On Storybook, at least with Firefox, when the menu is invisible, I'd expect clicking this would make it visible. But the menu stays visible only for a fraction of a second, then it closes and the focus moves into the input.

Reply:
Probably was broken because of the focus fuckery in the last comment you made, should be fixed now 👍